htdocs/takepos/invoice.php

```diff
@@ -165,8 +165,12 @@ function fail($message)
 	$constantforkey = 'CASHDESK_NO_DECREASE_STOCK'.$_SESSION["takeposterminal"];
 	if ($invoice->statut != Facture::STATUS_DRAFT)
 	{
-		dol_syslog("Sale already validated");
-		dol_htmloutput_errors($langs->trans("InvoiceIsAlreadyValidated", "TakePos"), null, 1);
+		//If invoice is validated but it is not fully paid is not error and make the payment
+		if ($invoice->getRemainToPay()>0) $res=1;
```

Comment:
This method is called here and later (line 206).
Because it includes some sql data, it should be better to make it once only.
Can you make here
$remaintopay = $invoice->getRemainToPay();
if ($remaintopay) $res=1;